alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/healthprocessor/fixer/solr/SolrMissingNodeFixerPlugin.java

```diff
@@ -1,15 +1,21 @@
 package eu.xenit.alfresco.healthprocessor.fixer.solr;
 
 import eu.xenit.alfresco.healthprocessor.fixer.api.NodeFixReport;
+import eu.xenit.alfresco.healthprocessor.fixer.api.NodeFixStatus;
 import eu.xenit.alfresco.healthprocessor.plugins.solr.NodeIndexHealthReport;
 import eu.xenit.alfresco.healthprocessor.plugins.solr.NodeIndexHealthReport.IndexHealthStatus;
 import eu.xenit.alfresco.healthprocessor.plugins.solr.SolrRequestExecutor;
 import eu.xenit.alfresco.healthprocessor.plugins.solr.SolrRequestExecutor.SolrNodeCommand;
+import eu.xenit.alfresco.healthprocessor.plugins.solr.endpoint.SearchEndpoint;
 import eu.xenit.alfresco.healthprocessor.reporter.api.NodeHealthReport;
+import org.alfresco.service.cmr.repository.NodeRef;
+
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.Set;
 
 public class SolrMissingNodeFixerPlugin extends AbstractSolrNodeFixerPlugin {
+    private HashMap<SearchEndpoint, HashMap<Long, NodeFixReport>> searchEndpointTxCache = new HashMap<>();
```

Comment:
Instead of a nested hashmap, can you create a simple value-object containing both search endpoint & txnid? That will make lookups & stores easier, as there is no need to manage a 2 level Map structure.

Additionally, use `Map<>` instead of `HashMap<>`